src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -838,6 +838,8 @@ public GitClient createClient(TaskListener listener, EnvVars environment, Run<?,
 
         String gitExe = getGitExe(n, listener);
 
+        GitTool gitTool = getGitTool(n, null, listener);
```

Comment:
This was added to access the `GitTool` chosen by the system instead of the git executable path for GitToolChooser.